packages/fast-components-react-base/src/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,276 @@
+
+        /**
+         * If we focus outside of the tree
+         */
+        if (isHTMLElement(root) && !root.contains(e.relatedTarget as HTMLElement)) {
+            this.setState({
+                focusable: true,
+            });
+        }
+    };
+
+    private handleFocus = (e: React.FocusEvent<HTMLDivElement>): void => {
+        if (!canUseDOM() || !isHTMLElement(this.rootElement.current)) {
+            return;
+        }
+
+        const root: HTMLElement | null = this.rootElement.current;
+        const lastFocused: { current: HTMLElement | null } | null = this.state
+            .lastFocused;
+
+        /**
+         * If the tree view is receiving focus
+         */
+        if (isHTMLElement(root) && root === e.target) {
+            // If we have a last focused item, focus it - otherwise check for an initially selected item or focus the first "[role='treeitem']"
+            // If there is no "[role='treeitem']" to be focused AND no last-focused, then there are likely no children
+            // or children are malformed so keep the tree in the tab-order in the hopes that the author cleans up
+            // the children
+            const selectedChild: HTMLElement | null = root.querySelector(
+                "[aria-selected='true']"
+            );
+
+            const toBeFocused: HTMLElement | null = !!lastFocused
+                ? lastFocused.current
+                : !!selectedChild
+                    ? selectedChild
+                    : root.querySelector("[role='treeitem']");
+
+            if (isHTMLElement(toBeFocused)) {
+                toBeFocused.focus();
+
+                if (this.state.focusable) {
+                    this.setState({ focusable: false });
+                }
+            }
+        } else {
+            // A child is receiving focus. While focus is within the tree, we simply need to ensure
+            // that the tree is not focusable.
+            if (this.state.focusable) {
+                this.setState({
+                    focusable: false,
+                });
+            }
+        }
+    };
+
+    /**
+     * Verifies that the tree has a focusable child.
+     * If it does not, the tree will begin to accept focus
+     */
+    private ensureFocusability(): void {
+        if (
+            canUseDOM() &&
+            !this.state.focusable &&
+            isHTMLElement(this.rootElement.current)
+        ) {
+            const focusableChild: HTMLElement | null = this.rootElement.current.querySelector(
+                "[role='treeitem'][tabindex='0']"
+            );
+
+            if (!isHTMLElement(focusableChild)) {
+                this.setState({
+                    focusable: true,
+                });
+            }
+        }
+    }
+
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        let nodes: HTMLElement[] | void;
+
+        if (canUseDOM()) {
+            switch (e.keyCode) {
+                case keyCodeHome:
```

Comment:
Prevent default on key events I think.

Reply:
Do we need to prevent default on *all* potential key events? I'm not sure we do

Reply:
If this is currently working as expected I would hesitate to change it.